docker/entrypoint.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/sh
+
+# usage: file_env VAR [DEFAULT]
+#    ie: file_env 'XYZ_DB_PASSWORD' 'example'
+# (will allow for "$XYZ_DB_PASSWORD_FILE" to fill in the value of
+#  "$XYZ_DB_PASSWORD" from a file, especially for Docker's secrets feature)
+file_env() {
+    local var="$1"
+    local fileVar="${var}_FILE"
+    local def="${2:-}"
+    if [ "${!var:-}" ] && [ "${!fileVar:-}" ]; then
+        echo >&2 "error: both $var and $fileVar are set (but are exclusive)"
+        exit 1
+    fi
+    local val="$def"
+    if [ "${!var:-}" ]; then
+        val="${!var}"
+    elif [ "${!fileVar:-}" ]; then
+        val="$(< "${!fileVar}")"
+    fi
+    export "$var"="$val"
+    unset "$fileVar"
+}
+
+file_env 'IOTA_CB_HOST'
+file_env 'IOTA_CB_PORT'
+file_env 'IOTA_MONGO_HOST'
+file_env 'IOTA_MONGO_PORT'
+file_env 'IOTA_AUTH_USER'
+file_env 'IOTA_AUTH_PASSWORD'
+
+if [[ $IOTA_AUTH_USER ==  "iotagent" ]] || [[ $IOTA_AUTH_PASSWORD == "iotagent'" ]]; then
```

Comment:
The `echo` here should do **one** of the following:

1. If we're running without protection echo a warning.
2. If we're running with protection but haven't overridden the conf ENV variables, echo a warning (it will  probably fail to connect)
3. If we're running with protection, have overridden the conf ENV, but have used plain text ENVs, echo a warning

In otherwords encourage good practice, but don't spam the user with irrelevant start up messages.

Reply:
Fixed with https://github.com/telefonicaid/iotagent-json/pull/380/commits/df770138c08a0de85d4a79a673af0d1477ec4d8e